model/project_ref.go

```diff
@@ -530,6 +518,171 @@ func recursivelySetUndefinedFields(structToSet, structToDefaultFrom reflect.Valu
 	}
 }
 
+func setRepoFieldsFromProjects(repoRef *RepoRef, projectRefs []ProjectRef) {
+	if len(projectRefs) == 0 {
+		return
+	}
+	reflectedRepo := reflect.ValueOf(repoRef).Elem().Field(0) // specifically references the ProjectRef part of RepoRef
+	for i := 0; i < reflectedRepo.NumField(); i++ {
+		// for each field in the repo, look at each field in the project ref
+		var firstVal reflect.Value
+		allEqual := true
+		for j, pRef := range projectRefs {
+			reflectedBranchField := reflect.ValueOf(pRef).Field(i)
+			if j == 0 {
+				firstVal = reflectedBranchField
+			} else if !reflect.DeepEqual(firstVal.Interface(), reflectedBranchField.Interface()) {
+				allEqual = false
+				break
+			}
+		}
+		// if we got to the end of the loop, then all values are the same, so we can assign it to reflectedRepo
+		if allEqual {
+			reflectedRepo.Field(i).Set(firstVal)
+		}
+	}
+}
+
+func (p *ProjectRef) createNewRepoRef(u *user.DBUser) (repoRef *RepoRef, err error) {
+	repoRef = &RepoRef{ProjectRef{
+		Id:      mgobson.NewObjectId().Hex(),
+		Owner:   p.Owner,
+		Repo:    p.Repo,
+		Enabled: utility.TruePtr(),
+		Admins:  []string{},
+	}}
+
+	allEnabledProjects, err := FindMergedEnabledProjectRefsByOwnerAndRepo(p.Owner, p.Repo)
+	if err != nil {
+		return nil, errors.Wrap(err, "error finding all enabled projects")
+	}
+	// for every setting in the project ref, if all enabled projects have the same setting, then use that
+	defer func() {
+		err = recovery.HandlePanicWithError(recover(), err, "project and repo structures do not match")
+	}()
+	setRepoFieldsFromProjects(repoRef, allEnabledProjects)
```

Comment:
Does it make sense to also unset these fields in the project refs? After doing this the field will be set in both the repo and branches which isn't ideal because the repo field won't do anything

Reply:
What if turning off the repo populates each branch with the setting, if it's not overridden? Having both populated just seems like it's encouraging bad configuration

Reply:
Well for one thing, setRepoFieldsFromProjects uses _all_ enabled projects, not just the one that's deciding to use repo ref, so it _would_ need to be populated for both for every project but that one, since the rest aren't using the repo yet. I also really don't prefer the idea of silently removing things from the project they intend to edit, with removals they didn't authorize. I believe stakeholders will turn this on initially to see what it looks like, but won't want to rely on it for every setting, just a few for testing. We should give them that control.

Reply:
Well it wouldn't be silent - I think a user could reasonably expect that turning on using the repo means that common settings would now be controlled by the repo. I suppose this is easy enough to revisit afterwards so it doesn't need to block this pr

Reply:
I'll make a comment in the ticket so we don't forget to think about it.